src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -1577,9 +1577,7 @@ void enableSSL(String host,
         String tmfDefaultAlgorithm = null;  // Default algorithm (typically X.509) used by the TrustManagerFactory
         SSLHandhsakeState handshakeState = SSLHandhsakeState.SSL_HANDHSAKE_NOT_STARTED;
 
-        boolean isFips = false;
         String trustStoreType = null;
-        String fipsProvider = null;
```

Comment:
@ulvi :

In this case as FIPSProvider is non-FIPS it's not fip connection. One can test it by [FipsEnvTest#isFips](https://github.com/Microsoft/mssql-jdbc/blob/master/src/test/java/com/microsoft/sqlserver/jdbc/fips/FipsEnvTest.java#L142)
> ds.setTrustStoreType("JCEKS");
ds.setFIPS(true);
ds.setFIPSProvider("IBMJCE");  //Not IBMJCEFIPS
connection = (SQLServerConnection) ds.getConnection();

For one of user I forked out SSLProtocol which having 2 fixes: 
1. Configurable SSL Protocol (Which you already worked on)
2. Made FIPSProvider optional as IBMFIPS not like FIPSProvider while calling `Keystore.getInstance` . (Need to test what happens in JDK 9 & Bouncy Castle for configurable configuraions)

Unfortunately, we did not finalize on solution with user, but this worked on user environment with appropriate SSL certificate.

If you use my  [SSLProtocol fork](https://github.com/Microsoft/mssql-jdbc/pull/325) and use following code for IBM FIPS.

```
//ds.setTrustStoreType("JCEKS"); (As per my understanding IBM works well with JKS too in FIPS)
ds.setFIPS(true);
//ds.setFIPSProvider("IBMJCE");  For IBM FIPS one should not set FIPSProvider.
connection = (SQLServerConnection) ds.getConnection();
```
Possibly @v-xiangs might have some information. Feel free to call me.